crates/wasmparser/src/validator/operators.rs

```diff
@@ -524,6 +534,12 @@ impl<'resources, R: WasmModuleResources> OperatorValidatorTemp<'_, 'resources, R
         };
         let ty = frame.block_type;
         let height = frame.height;
+        let init_height = frame.init_height;
+
+        // reset_locals in the spec
+        for init in self.inits.split_off(init_height) {
+            self.local_inits[init as usize] = false;
+        }
```

Comment:
I suppose stuff like this should only really be run if `function_references` are enabled -- though, I suppose it might be close to an noop if they aren't as `init_height = 0` then.

Reply:
Yes, because all value types without function references are defaultable.    So, we could disable initializing the stacks, but that only happens (number of functions) times.    For our purposes it's certainly fine and for upstream we can see if Alex complains?   :)